### PR TITLE
Reference states sanctions body Cordapp Migrated to 4.12-RC01 and JDK17

### PR DESCRIPTION
**Pull request by ashwani-r3**

### CODE REVIEW

File: Features/constants.properties [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
-gradlePluginsVersion=5.0.12
-kotlinVersion=1.2.71
-junitVersion=4.12
-quasarVersion=0.7.10
-log4jVersion =2.17.1
-platformVersion=10
-slf4jVersion=1.7.25
-nettyVersion=4.1.22.Final
+cordaVersion=4.12-RC01
+cordaCoreVersion=4.12-RC01
```
Comment (adelel1) [inline Features/constants.properties:4]: This can be RC02 now

File: Features/referenceStates-sanctionsBody/build.gradle [hunk 2]
```diff
@@ -11,14 +13,47 @@ buildscript {
...
         corda_platform_version = constants.getProperty("platformVersion").toInteger()
+        spring_boot_version = '3.2.5'
+        spring_boot_gradle_plugin_version = '3.2.5'
+        testJvmArgs = ['--add-opens', 'java.base/java.time=ALL-UNNAMED', '--add-opens', 'java.base/java.io=ALL-UNNAMED',
+                       '--add-opens', 'java.base/java.util=ALL-UNNAMED', '--add-opens', 'java.base/java.net=ALL-UNNAMED',
+                       '--add-opens', 'java.base/java.nio=ALL-UNNAMED', '--add-opens', 'java.base/java.lang.invoke=ALL-UNNAMED',
+                       '--add-opens', 'java.base/java.security.cert=ALL-UNNAMED', '--add-opens', 'java.base/java.security=ALL-UNNAMED',
+                       '--add-opens', 'java.base/javax.net.ssl=ALL-UNNAMED', '--add-opens', 'java.base/java.lang=ALL-UNNAMED',
+                       '--add-opens', 'java.base/java.util.concurrent=ALL-UNNAMED', '--add-opens', 'java.sql/java.sql=ALL-UNNAMED',]
+        testJvmFxArgs = [  '--add-exports', 'java.base/sun.nio.ch=ALL-UNNAMED',
```
Comment (jakubzadroga) [inline Features/referenceStates-sanctionsBody/build.gradle:48]: Is this needed? Can't see it used anywhere in this CorDapp

**Review by jakubzadroga — CHANGES_REQUESTED** [review]
Two comments please

**Review by jakubzadroga — APPROVED** [review]
LGTM

**Pull request merged**